rotkehlchen/api/rest.py

```diff
@@ -4062,3 +4062,38 @@ def import_user_assets(self, path: Path) -> Response:
             )
 
         return api_response(OK_RESULT, status_code=HTTPStatus.OK)
+
+    @require_loggedin_user()
+    def export_user_db_snapshot(self, timestamp: Timestamp) -> Response:
+        timed_location_data = self.rotkehlchen.snapshot.get_timed_location_data(timestamp)
+        timed_balances = self.rotkehlchen.snapshot.get_timed_balances(timestamp)
+
+        is_success, message = self.rotkehlchen.snapshot.export(timed_balances, timed_location_data)
+        if is_success is False:
+            return api_response(wrap_in_fail_result(message), status_code=HTTPStatus.CONFLICT)
+
+        return api_response(OK_RESULT, status_code=HTTPStatus.OK)
+
+    @require_loggedin_user()
+    def download_user_db_snapshot(self, timestamp: Timestamp) -> Response:
+        timed_location_data = self.rotkehlchen.snapshot.get_timed_location_data(timestamp)
+        timed_balances = self.rotkehlchen.snapshot.get_timed_balances(timestamp)
+
+        is_success, zipfile_path = self.rotkehlchen.snapshot.export(
+            timed_balances, timed_location_data,
+        )
```

Comment:
If you do what I told you this should be replaced by

```suggestion
       dbsnapshots = DBSnapshot(self.rotkehlchen.data.db)
       is_success, zipfile_path = snapshot.export(timestamp=timestamp, path=None)
```